test/model/store_test.dart

```diff
@@ -732,7 +732,12 @@ void main() {
       });
     }
 
-    test('retries on NetworkException', () {
+    test('retries on NetworkException from SocketException', () {
+      // We skip reporting errors on these; check we retry them all the same.
```

Comment:
I see that later we add a test to verify that we indeed ignore such errors, maintaining some sort of "each error that we retries/reloads on gets a corresponding 'report error' test" relationship. But there is this visual gap between each pair of them. Is it feasible to place them next to each other?

Reply:
Hmm, this is meant to coincide with one of the error types that in main we already test that we skip reporting:
```dart
      test('ignore boring errors', () => awaitFakeAsync((async) async {
        await prepare();

        for (int i = 0; i < UpdateMachine.transientFailureCountNotifyThreshold; i++) {
          connection.prepare(exception: const SocketException('failed'));
          pollAndFail(async);
          check(takeLastReportedError()).isNull();
```

Reply:
Yeah, this is quite a nit. I was referring to the visual gap between this test and
```dart
      test('ignore NetworkException from SocketException', () {
        checkNotReported(prepareNetworkExceptionSocketException);
      });
```
or more in general, the gap between the 'report error` group and the tests before it. We have 10 such pairs:

```dart
    test('reloads on unexpected error within loop', () {
      checkReload(prepareUnexpectedLoopError);
    });

	// ...

    group('report error', () {

	  // ...

      test('report unexpected error within loop', () {
        checkReported(prepareUnexpectedLoopError);
      });
```

I'm not sure if it would be better to group the tests for the same error close to each other, or keep it this way. It does seem that such tests tend to co-occur.

Reply:
I see, yeah.

There's a fundamental trade-off to make: these test cases are really a matrix, of (which type of error) x (what to do vs. what to report). So we can group the different checks on a given error type together, but then the different error types get spread farther apart, or vice versa.

I went for this arrangement — different error types together for each kind of check — because I think when considering what we want to do (or report) for a given error type it's helpful to be able to compare across other error types. With this arrangement, all the reload/retry tests fit on the screen at once, or a couple of screenfuls on a smaller display.

Reply:
That's a great way to put it! Thanks for the explanation.